docs/Quick_start_guide.rst

```diff
@@ -0,0 +1,144 @@
+===========================
+Quick start guide
+===========================
+
+Usage
+-----
+
+Basic examples
+..............
+
+Here is a simple python script that reads an image, applies a median image filter (radius of 2 pixels), and writes the resulting image in a file.
+
+  ::
+
+    import itk
+
+    image = itk.imread("/path/to/my/image.png")
+    median = itk.MedianImageFilter.New(image, Radius = 2)
+    itk.imwrite(median, "path/to/my/output_image.png")
+
+There are two ways to instantiate filters with ITKPython:
+
+- Implicit (recommended): ITK type information is automatically detected from the data. Under the hood typed filter objects and images are created. 
+
+  ::
+
+    # Use `ImageFileReader` instead of the wrapping function `imread` to illustrate this example.
+    reader = itk.ImageFileReader.New(FileName="/path/to/my/image.png")
+    # Here we specify the filter input explicitly
+    median = itk.MedianImageFilter.New(Input=reader.GetOutput())
+    # Same as above but shortened. `Input` does not have to be specified.
+    median = itk.MedianImageFilter.New(reader.GetOutput())
+    # Same as above. `.GetOutput()` does not have to be specified.
+    median = itk.MedianImageFilter.New(reader)
+
+- Explicit: This can be useful if a filter cannot automatically select the type information (e.g. `CastImageFilter`), or to detect type mismatch errors which can lead to cryptic error messages.
+
+  - Explicit instantiation of median image filter
+
+  ::
+
+    ImageType = itk.Image[itk.F,2]
+    median = itk.MedianImageFilter[ImageType, ImageType].New()
+    median.SetInput(reader.GetOutput())
+
+  - Explicit instantiation of cast image filter
+
+  ::
+
+    import itk
+    import sys
+    reader = itk.imread("/path/to/my/image.png")
+    InputType = type(reader.GetOutput())
+    # Find input image dimension
+    InputDimension = reader.GetOutput().GetDimension()
+    # Select float as output pixel type
+    OutputType = itk.Image[itk.F, InputDimension]
+    castFilter = itk.CastImageFilter[InputType, OutputType].New()
+    castFilter.SetInput(reader)
+    itk.imwrite(castFilter, "path/to/my/output_image.png")
+
+ITK Pyton types
+...............
+
++---------------------+--------------------+
+| C++ type            | Python type        |
++=====================+====================+
+| float               | itk.F              |
++---------------------+--------------------+
+| double              | itk.D              |
++---------------------+--------------------+
+| unsigned char       | itk.UC             |
++---------------------+--------------------+
+| std::complex<float> | itk.complex[itk.F] |
++---------------------+--------------------+
+
+This list is not exhautive and is only presented to illustrate the type names.
```

Comment:
May be a ITK Python Style section could be added

something like

```
Coding Style
------------

* Type variable: ITK types should be specified as `ImageType` instead of `image_type`. See https://www.python.org/dev/peps/pep-0008/#type-variable-names
```